config/routes.rb

```diff
@@ -26,7 +26,8 @@
   get '/users/:id/medicine_notebook/new', to: 'medicine_notebook_records#new', as: :medicine_notebook_new
   post '/users/:id/medicine_notebook/new', to: 'medicine_notebook_records#create'
   get '/users/:id/medicine_notebook/show', to: 'medicine_notebook_records#show', as: :medicine_notebook_show
-  get '/users/:id/medicine_notebook/edit', to: 'medicine_notebook_records#edit'
-  post '/users/:id/medicine_notebook/edit', to: 'medicine_notebook_records#edit'
+  get '/users/:id/medicine_notebook/:medicine_notebook_id/edit', to: 'medicine_notebook_records#edit', as: :medicine_notebook_edit
+  patch '/users/:id/medicine_notebook/:medicine_notebook_id/edit', to: 'medicine_notebook_records#update'
```

Comment:
editは編集ページなので、更新のルーティングは下記で良さそうです（deleteも同様）
```suggestion
  patch '/users/:id/medicine_notebook/:medicine_notebook_id', to: 'medicine_notebook_records#update'
```